File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int l = 0;
        int r = height.size() -1;
        int ma = 0;
        int min = 0; int a= 0;
        while (l < r){
            if (height[l] < height[r]){
                min = height[l];
                l++;
            }
            else {
                min = height[r];
                r--;
            }
            ma = max(ma, (r - l + 1) * min);
        }
        return ma;
    }
};

File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l, r = 0, len(height) - 1
        water = 0
        while l < r:
            if height[l] < height[r]:
                water = max(water, (r-l) * height[l])
                l += 1
            else:
                water = max(water, (r-l) * height[r])
                r -= 1
        return water
        